parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -387,17 +387,21 @@ ensureKubelet() {
 }
 
 ensureAddons() {
-  retrycmd 120 5 30 $KUBECTL get pods -l app=kube-addon-manager -n kube-system || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
+  if ! { [ -f /var/run/reboot-required ] || [ "$NO_OUTBOUND" = "true" ]; }; then
+    retrycmd 120 5 30 $KUBECTL get pods -l app=kube-addon-manager -n kube-system || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
 {{- if not HasCustomPodSecurityPolicy}}
-  retrycmd 120 5 30 $KUBECTL get podsecuritypolicy privileged restricted || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
-  rm -Rf ${ADDONS_DIR}/init
+    retrycmd 120 5 30 $KUBECTL get podsecuritypolicy privileged restricted || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
+    rm -Rf ${ADDONS_DIR}/init
```

Comment:
I'm not too sure what this does and whether we want to run it every time or also skip it

Reply:
I think your change is correct - we only want to clean up after the deployment.

Reply:
Only question I have is what should happen if we are in "need of reboot" - what additional work below is still acceptable?

(we should think/model this as a state machine)

Reply:
So what is happening here:

1) Cloud-init populates a subset of "must load first" addons (at present pod-security-policy) to `/etc/kubernetes/addons/init`
2) The remaining addons are in `/etc/kubernetes/addons/`
3) kube-addon-manager configuration says "source addons from the init directory"
4) we wait for kube-addon-manager pod to be Running (which is an indirect way of confirming that the "init" addons have been loaded)
5) we then move all addons into the main `/etc/kubernetes/addons` directory, update the kube-addon-manager configuration so that it sources from there (and no longer init) and then restart kube-addon-manager to load the remaining addons

So, if we're in a node reboot situation, we should ensure that we do step 5, but we can ignore the others.

Does that make sense?

Reply:
which step is `rm -Rf ${ADDONS_DIR}/init` part of?